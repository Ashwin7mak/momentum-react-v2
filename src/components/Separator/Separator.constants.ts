const CLASS_PREFIX = 'md-separator';

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, STYLE };
