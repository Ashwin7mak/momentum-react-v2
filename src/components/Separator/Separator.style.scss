.md-separator-wrapper {
  color: var(--theme-text-secondary-normal);
  font-family: 'CiscoSansTT Regular';

  div {
    display: flex;
    flex-direction: row;

    &:before,
    &:after {
      content: '';
      flex: 1 1;
      border-bottom: 1px solid var(--theme-outline-primary-normal);
      margin: auto;
    }

    &:before {
      margin-right: 0.5rem;
    }

    &:after {
      margin-left: 0.5rem;
    }
  }
}
