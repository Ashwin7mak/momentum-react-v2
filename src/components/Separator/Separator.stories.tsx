import React from 'react';
import { Story } from '@storybook/react';

import Separator, { SeparatorProps } from './';

import Icon from '../Icon';

const childrenOptions = {
  'Some separator text': 'Some separator text',
  icon: (
    <div>
      <Icon name="meetings" />
    </div>
  ),
};

export default {
  title: 'Momentum UI/Separator',
  component: Separator,
  argTypes: {
    children: {
      description: 'Text or other component to put in the middle of the separator',
      options: Object.keys(childrenOptions),
      mapping: childrenOptions,
      control: { type: 'select' },
    },
  },
};

const Template: Story<SeparatorProps> = (args) => {
  const { children } = args;
  return (
    <div style={{ width: '15rem' }}>
      <Separator {...args}>{children}</Separator>
    </div>
  );
};

const Story1 = Template.bind({});

Story1.args = {
  children: 'Some separator text',
};

export { Story1 };
