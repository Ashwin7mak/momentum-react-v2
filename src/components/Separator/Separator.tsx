import React from 'react';
import classnames from 'classnames';

import './Separator.style.scss';
import { Props } from './Separator.types';
import { STYLE } from './Separator.constants';

const Separator: React.FC<Props> = (props: Props) => {
  const { className, children } = props;

  return (
    <div className={STYLE.wrapper}>
      <div className={classnames(className)}>{children}</div>
    </div>
  );
};

/**
 * Separator with text/other component in the middle
 */

export default Separator;
