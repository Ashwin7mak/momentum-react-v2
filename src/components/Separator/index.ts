import { default as Separator } from './Separator';
import { Props } from './Separator.types';

export type SeparatorProps = Props;

export default Separator;
