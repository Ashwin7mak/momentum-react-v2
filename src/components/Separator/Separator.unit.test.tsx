import Separator from '.';
import { mount } from 'enzyme';
import React from 'react';

describe('Separator', () => {
  describe('snapshot', () => {
    it('should match snapshot when text is used', () => {
      const container = mount(<Separator>hello</Separator>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when another component is used', () => {
      const container = mount(
        <Separator>
          <div>
            <div>Some nested text</div>
          </div>
        </Separator>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
