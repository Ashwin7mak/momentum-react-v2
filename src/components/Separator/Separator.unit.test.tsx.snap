// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator snapshot should match snapshot when another component is used 1`] = `
<Separator>
  <div
    className="md-separator-wrapper"
  >
    <div
      className=""
    >
      <div>
        <div>
          Some nested text
        </div>
      </div>
    </div>
  </div>
</Separator>
`;

exports[`Separator snapshot should match snapshot when text is used 1`] = `
<Separator>
  <div
    className="md-separator-wrapper"
  >
    <div
      className=""
    >
      hello
    </div>
  </div>
</Separator>
`;
